client/src/components/Ticketing/ticketingmanager/ticketingSlice.ts

```diff
@@ -313,12 +338,54 @@ export const createCartItem = (data: {
       payWhatCan: tickettype.name === 'Pay What You Can',
     };
 
-    if (cartItem.payWhatCan) {
-      cartItem.price = payWhatPrice;
+    if (ticketCartItem.payWhatCan) {
+      ticketCartItem.price = payWhatPrice;
     }
 
-    return cartItem;
+    return ticketCartItem;
+  }
+};
+
+export interface SeasonSubscriptionType {
+    seasonid_fk: number;
+    subscriptiontypeid_fk: number;
+    subscriptionlimit: number;
+    subscriptionssold: number;
+    price: number;
+    name: string;
+    ticketlimit: number;
+    season: SeasonInfo;
+}
+
+export const createSubscriptionCartItem = (
+  type: SeasonSubscriptionType,
+): SubscriptionCartItem | undefined => {
+  const {
+    seasonid_fk,
+    subscriptiontypeid_fk,
+    subscriptionlimit,
+    subscriptionssold,
+    name,
+    ticketlimit,
+    season,
+    price,
+  } = type;
+  if (
+    !seasonid_fk ||
+    !subscriptiontypeid_fk
+  ) {
+    return;
   }
+  return {
+    seasonid_fk,
+    subscriptiontypeid_fk,
+    qtyAvailable: subscriptionlimit - subscriptionssold,
+    name: `${name} Subscription`,
+    desc: `${ticketlimit} shows for ${season.name}`,
```

Comment:
Also might want a condition on the `s` in shows

Reply:
Fixed